### PR TITLE
psitransfer: init at 2.0.1

### PR DESCRIPTION
**Pull request by hyshka**
###### Motivation for this change

I use this tool myself for self-hosted filesharing and wanted to contribute to the Nix ecosystem.

Issue: #123383

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

###### Feedback needed

I added myself as a maintainer in [this PR](https://github.com/NixOS/nixpkgs/pull/149095).

I am not sure if this should be in `/pkgs/servers/web-apps/` or `/pkgs/servers/`.

This was a PITA to get building but now that I have it running I might want to work on some configuration options as well.

### CODE REVIEW

File: pkgs/servers/psitransfer/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  combined = psitransfer.override rec {
+    # version is not defined in source package.json
+    # version must also be maintained in node-packages.json for node2nix
+    version = "2.0.1";
+
+    # override node2nix package src to pull pre-built release of same version
+    src = fetchzip {
+      url
+      = "https://github.com/psi-4ward/psitransfer/releases/download/v${version}/psitransfer-v${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/servers/psitransfer/default.nix:30]:
```suggestion
      url = "https://github.com/psi-4ward/psitransfer/releases/download/v${version}/psitransfer-v${version}.tar.gz";
```
should fix the highlighting here.

File: pkgs/servers/psitransfer/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      sha256 = "mfldWTVmfcIRa+1g8YDnQqem5YmrFRfCxJoitWcXvns=";
+      stripRoot = false;
+    };
+
+    meta = with lib; {
+      inherit version;
```
Comment (SuperSandro2000) [inline pkgs/servers/psitransfer/default.nix:36]:
```suggestion
```
That is legacy. Also we should set description and license.

Comment (hyshka) [reply pkgs/servers/psitransfer/default.nix]: @SuperSandro2000 I removed the version here, this hasn't been updated in the CONTRIBUTING.md yet.

I added description and license as well, though those have been defined by `node2nix` in `node-packages.nix`.

Comment (SuperSandro2000) [reply pkgs/servers/psitransfer/default.nix]: > I also added myself as the maintainer which would be pending on this PR.

This shouldn't be done in another PR but in this one.

Comment (hyshka) [reply pkgs/servers/psitransfer/default.nix]: Thanks again @SuperSandro2000, I will remember that for next time.

**Pull request merged**